src/Arcus.Messaging.Pumps.ServiceBus/Configuration/AzureServiceBusMessagePumpConfiguration.cs

```diff
@@ -1,43 +1,42 @@
 ﻿using System;
 using GuardNet;
 
-namespace Arcus.Messaging.Pumps.ServiceBus
+namespace Arcus.Messaging.Pumps.ServiceBus.Configuration
 {
     /// <summary>
     ///     Options to configure how the Azure Service Bus message pump works
     /// </summary>
-    public class AzureServiceBusMessagePumpOptions
+    internal class AzureServiceBusMessagePumpConfiguration
     {
-        private int? _maxConcurrentCalls;
-
         /// <summary>
-        ///     Maximum concurrent calls to process messages
+        /// Initializes a new instance of the <see cref="AzureServiceBusMessagePumpConfiguration"/> class.
         /// </summary>
-        public int? MaxConcurrentCalls
+        internal AzureServiceBusMessagePumpConfiguration(AzureServiceBusQueueMessagePumpOptions options)
         {
-            get => _maxConcurrentCalls;
-            set
-            {
-                if (value != null)
-                {
-                    Guard.For<ArgumentException>(() => value <= 0, "Max concurrent calls has to be 1 or above.");
-                }
-
-                _maxConcurrentCalls = value;
-            }
+            Guard.NotNull(options, nameof(options));
```

Comment:
Should we guard on:
- jobId being empty or whatever
- max concurrent calls < 0 (might just rely on the SDK for this though)

Reply:
Hmmm, that is already check while we create the Topic/Queue options...